### PR TITLE
fix(permission): fail to submit ticket if lack of database permission

### LINKED ISSUE #2915: [Bug]: Table change permission cannot take effect
### ODC version

odc431

### OB version

*

### What happened?

image
image
image

### What did you expect to happen?

Tables with table change permissions can perform database changes

### How can we reproduce it (as minimally and precisely as possible)?

Low privilege account application form change permission, execute database change

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:

If  a user is granted some tables' permission without database permission, he could not sumit database change ticket.

#### Which issue(s) this PR fixes:

Fixes #2915

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/permission/DBResourcePermissionHelper.java [hunk 1]
```diff
@@ -425,7 +425,7 @@ private Map<Long, Set<DatabasePermissionType>> getInnerDBPermissionTypes(Collect
                 .findNotExpiredByUserIdAndDatabaseIdIn(authenticationFacade.currentUserId(), databaseIds)
                 .stream().collect(Collectors.toMap(
                         UserTablePermissionEntity::getDatabaseId,
-                        e -> Collections.singleton(DatabasePermissionType.ACCESS),
+                        e -> Collections.singleton(DatabasePermissionType.from(e.getAction())),
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/permission/DBResourcePermissionHelper.java:428]: group by will be better, cause that there will be multi tables that have the same database id

Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/permission/DBResourcePermissionHelper.java]: done~

**Review by yhilmare — APPROVED** [review]
LGTM

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**